### PR TITLE
feat: grpc requests methods

### PR DESCRIPTION
**Pull request by ZixMai**
1. Added async grpc requests methods 
2. Name changes in protobuf file
3. Added event repository arg in event service impl method in main
4. Changed docstrings
5. Added inheritance in EventRepositoryImpl class
6. Added and sorted some imports
7. Added logging to postgres_client.py
8. Small changes in arguments format

### CODE REVIEW

**Review by MaxPlays35 — APPROVED** [review]
Approved

File: event_service/repository/event_repository_impl.py [hunk 1]
```diff
@@ -65,7 +73,99 @@ async def get_events(self, author_id: str) -> List[Event]:
...
+        Raises
+        ------
+        prisma.errors.PrismaError
+            Catch all for every exception raised by Prisma Client Python.
+        ValueNotFoundError
+            No events was found for given event id.
+
+        """
+        db_event: Optional[PrismaEvent] = await self._db_client.db.event.find_first(
+            where={"id": event_id}
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Check that deleted at is None

File: event_service/repository/event_repository_impl.py [hunk 2]
```diff
@@ -65,7 +73,99 @@ async def get_events(self, author_id: str) -> List[Event]:
...
+        prisma.errors.PrismaError
+            Catch all for every exception raised by Prisma Client Python.
+        ValueNotFoundError
+            No events was found for given events ids.
+
+        """
+        db_events: Optional[
+            List[PrismaEvent]
+        ] = await self._db_client.db.event.find_many(
+            where={"id": {"in": [event_id for event_id in events_ids.ids]}}
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Check that deleted at is None

File: event_service/repository/mock_event_repository.py [hunk 3]
```diff
@@ -59,50 +66,137 @@ async def get_events(self, author_id: str) -> List[Event]:
...
+
+        """
+        try:
+            event = next(event for event in self._events if event.id == event_id)
+            return event
+        except StopIteration:
+            raise ValueNotFoundError("Events not found")
+
+    async def get_events_by_events_ids(
+        self, events_ids: ListOfEventsIds
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: Make this method accept event_ids: List[str]

File: event_service/repository/mock_event_repository.py [hunk 4]
```diff
@@ -59,50 +66,137 @@ async def get_events(self, author_id: str) -> List[Event]:
...
+        List[Event]
+            List of events that matches by event id.
+
+        Raises
+        ------
+        ValueNotFoundError
+            No events was found for given author id.
+
+        """
+        if self._events is None or len(self._events) == 0:
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: self._events should not be none

File: event_service/repository/mock_event_repository.py [hunk 5]
```diff
@@ -59,50 +66,137 @@ async def get_events(self, author_id: str) -> List[Event]:
...
+        ValueNotFoundError
+            Can't update event with provided data.
 
         """
-        self._events[self._events.index(event)] = event
+        try:
+            index = next(
+                i for i in range(len(self._events)) if self._events[i].id == event.id
+            )
+            if self._events[index].author_id == event.author_id:
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py:184]: This check is useless

File: event_service/src/event_service_impl.py [hunk 6]
```diff
@@ -54,9 +63,126 @@ def get_events(
...
+        EventResponse
+            Response object for event response.
+
+        """
+        try:
+            event = await self._event_repository.get_event_by_event_id(
+                event_id=request.event_id
+            )
+            context.set_code(grpc.StatusCode.OK)
+            return proto.EventResponse(status_code=200, event=event)
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: should be event= event.to_grpc_event()

File: event_service/src/event_service_impl.py [hunk 7]
```diff
@@ -54,9 +63,126 @@ def get_events(
...
 
         """
-        pass
+        try:
+            events = await self._event_repository.get_events_by_author_id(
+                author_id=request.author_id
+            )
+            context.set_code(grpc.StatusCode.OK)
+            return proto.EventsResponse(
+                status_code=200, events=proto.ListOfEvents(events=events)
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: should use generator to convert events to grpc events

File: event_service/src/event_service_impl.py [hunk 8]
```diff
@@ -54,9 +63,126 @@ def get_events(
...
+            Response object for event response.
+
+        """
+        try:
+            events = await self._event_repository.get_events_by_events_ids(
+                events_ids=request.events_ids
+            )
+            context.set_code(grpc.StatusCode.OK)
+            return proto.EventsResponse(
+                status_code=200, events=proto.ListOfEvents(events=events)
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: should use generator to convert events to grpc events

File: event_service/src/event_service_impl.py [hunk 9]
```diff
@@ -54,9 +63,126 @@ def get_events(
...
+        -------
+        EventsResponse
+            Response object for event response.
+
+        """
+        try:
+            events = await self._event_repository.get_all_events()
+            context.set_code(grpc.StatusCode.OK)
+            return proto.EventsResponse(
+                status_code=200, events=proto.ListOfEvents(events=events)
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: should use generator to convert events to grpc events

File: event_service/src/models/event.py [hunk 10]
```diff
@@ -90,7 +90,7 @@ def to_dict(self, exclude: Optional[List[str]] = None) -> Dict[str, Any]:
             Event data dictionary.
 
         """
-        exclude_set = set(exclude if exclude is not None else ["id"])
+        exclude_set = set(exclude if exclude is not None else []) | {"id", "author_id"}
```
Comment (Nhsdkk) [inline event_service/src/models/event.py]: no need to exclude author_id

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
check the review and mate changes accordingly

**Comment by Nhsdkk** [comment]
When you get events you should return all event data, including created_at and deleted_at, and also this field should be modifiable in update_event

File: event_service/src/event_service_impl.py [hunk 11]
```diff
@@ -35,15 +45,125 @@ def __init__(
...
+
+        Returns
+        -------
+        EventsResponse
+            Response object for event response.
+
+        """
+        try:
+            events = await self._event_repository.get_events_by_events_ids(
+                events_ids=[event_id for event_id in request.events_ids.ids]
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: Just write list(request.events_ids.ids)

File: event_service/src/models/event.py [hunk 12]
```diff
@@ -122,6 +132,47 @@ def from_prisma_event(cls, prisma_event: PrismaEvent) -> Self:
...
+            description=grpc_event.description,
+            color=grpc_event.color,
+            repeating_delay=datetime.fromtimestamp(
+                grpc_event.repeating_delay.seconds
+                + grpc_event.repeating_delay.nanos / 1e9
+            ),
+            created_at=datetime.fromtimestamp(
+                grpc_event.created_at.seconds + grpc_event.created_at.nanos / 1e9
+            ),
+            deleted_at=datetime.fromtimestamp(
```
Comment (Nhsdkk) [inline event_service/src/models/event.py:178]: check for None

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Check and make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**